src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -403,16 +403,22 @@ int uct_ib_mlx5_devx_uar_cmp(uct_ib_mlx5_devx_uar_t *uar,
 #if HAVE_DEVX
 static ucs_status_t
 uct_ib_mlx5_devx_alloc_uar(uct_ib_mlx5_md_t *md, unsigned flags, int log_level,
-                           char *title, char *comment,
+                           char *title, char *fallback,
                            struct mlx5dv_devx_uar **uar_p)
 {
     struct mlx5dv_devx_uar *uar;
+    char buf[512];
 
     uar = mlx5dv_devx_alloc_uar(md->super.dev.ibv_context, flags);
     if (uar == NULL) {
-        ucs_log(log_level, "mlx5dv_devx_alloc_uar(device=%s, flags=0x%x(%s)) "
-                "failed: %m%s", uct_ib_device_name(&md->super.dev), flags,
-                title, comment);
+        sprintf(buf, "mlx5dv_devx_alloc_uar(device=%s, flags=0x%x(%s)) "
+                "failed: %m", uct_ib_device_name(&md->super.dev), flags, title);
+        if (fallback == NULL) {
+            ucs_log(log_level, "%s", buf);
+        } else {
+            ucs_log(log_level, "%s, fallback to %s", buf, fallback);
+        }
+
```

Comment:
what was wrong with the previous version?

Reply:
coverity